packages/google_sign_in/google_sign_in_ios/ios/Classes/FLTGoogleSignInPlugin.m

```diff
@@ -139,19 +121,36 @@ - (nullable NSNumber *)isSignedInWithError:
 - (void)signInWithCompletion:(nonnull void (^)(FSIUserData *_Nullable,
                                                FlutterError *_Nullable))completion {
   @try {
-    GIDConfiguration *configuration = self.configuration
-                                          ?: [self configurationWithClientIdArgument:nil
-                                                              serverClientIdArgument:nil
-                                                                hostedDomainArgument:nil];
-    [self.signIn signInWithConfiguration:configuration
-                presentingViewController:[self topViewController]
-                                    hint:nil
-                        additionalScopes:self.requestedScopes.allObjects
-                                callback:^(GIDGoogleUser *user, NSError *error) {
-                                  [self didSignInForUser:user
-                                          withCompletion:completion
-                                                   error:error];
-                                }];
+    // If the configuration settings are passed from the Dart API, use those.
+    // Otherwise, use settings from the GoogleService-Info.plist if available.
+    // If neither are available, do not set the configuration - GIDSignIn will automatically use
+    // settings from the Info.plist (which is the recommended method).
+    if (!self.configuration && self.googleServiceProperties) {
+      self.configuration = [self configurationWithClientIdArgument:nil
+                                            serverClientIdArgument:nil
+                                              hostedDomainArgument:nil];
+    }
+    if (self.configuration) {
+      self.signIn.configuration = self.configuration;
+    }
+
+    [self.signIn signInWithPresentingViewController:[self topViewController]
+                                               hint:nil
+                                   additionalScopes:self.requestedScopes.allObjects
+                                         completion:^(GIDSignInResult *_Nullable signInResult,
+                                                      NSError *_Nullable error) {
+                                           GIDGoogleUser *user;
+                                           NSString *serverAuthCode;
+                                           if (signInResult) {
+                                             user = signInResult.user;
+                                             serverAuthCode = signInResult.serverAuthCode;
+                                           }
+
+                                           [self didSignInForUser:user
+                                                   serverAuthCode:serverAuthCode
+                                                   withCompletion:completion
```

Comment:
nit: the `with` should go on the first additional argument, so `didSignInForUser:withServerAuthCode:completion:error:`.